scipy/spatial/transform/rotation.py

```diff
@@ -84,3 +85,41 @@ def as_quaternion(self):
             return self._quat[0]
         else:
             return self._quat
+
+    def as_dcm(self):
+        """Return the direction cosine matrix representation of the Rotation.
+
+        This function returns a numpy.ndarray of shape (3 x 3) or (N x 3 x 3)
+        depending on the input that was used to initialize the object.
+        """
+        quat = self._quat.copy()
+
+        x = quat[:, 0].reshape(-1, 1)
+        y = quat[:, 1].reshape(-1, 1)
+        z = quat[:, 2].reshape(-1, 1)
+        w = quat[:, 3].reshape(-1, 1)
+
+        x2 = np.power(x, 2)
+        y2 = np.power(y, 2)
+        z2 = np.power(z, 2)
+        w2 = np.power(w, 2)
+
+        xy = x * y
+        zw = z * w
+        xz = x * z
+        yw = y * w
+        yz = y * z
+        xw = x * w
+
+        d1 = np.concatenate(
```

Comment:
OK, I suggest a simpler approach. I think you will like it more.

1. Create `dcm` with as `np.empty` with an appropriate shape.
2. Assign `dcm[:, 0, 0]`, `dcm[:, 0, 1]`, etc. This way you also don't need to do any broadcasting for quaternion components.

I think it will make the conversion procedure easier to understand (as implementation reference) and it will be slightly more efficient as well (including memory allocation).